src/coreclr/tools/Common/TypeSystem/Common/CallingConventions.cs

```diff
@@ -0,0 +1,195 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Immutable;
+using System.Reflection.Metadata;
+
+using Internal.TypeSystem.Ecma;
+
+using Debug = System.Diagnostics.Debug;
+
+namespace Internal.TypeSystem
+{
+    /// <summary>
+    /// Enumeration that represents a unified view of calling conventions
+    /// irrespective of their metadata encoding.
+    /// </summary>
+    [Flags]
+    public enum CallingConventions
+    {
+        //
+        // Calling convention modifiers
+        //
+        ModifiersMask           = unchecked((int)0xFF000000),
+        IsSuppressGcTransition  = 0x01000000,
+        IsMemberFunction        = 0x02000000,
+
+        //
+        // Calling conventions
+        //
+        CallingConventionMask   = 0x00FFFFFF,
+
+        // Keep the ones defined in MethodSignatureFlags bitcastable
+        Cdecl                   = 0x00000001,
+        Stdcall                 = 0x00000002,
+        Thiscall                = 0x00000003,
+        Fastcall                = 0x00000004,
+        Varargs                 = 0x00000005,
+
+        /// <summary>
+        /// Platform-default unmanaged calling convention.
+        /// </summary>
+        Unmanaged               = 0x00000009,
```

Comment:
On a second thought, this should translate to whatever is the platform default. We shouldn't have this enum member.